### PR TITLE
Improve documentation

### PR DESCRIPTION
**Pull request by Uzlopak**
Improving the documentation. 

I added the recommendation to supply a custom getToken function as module option. I think it is safer and more performance to do that.

I think this is a review task for @Fdawgs ;)

#### Checklist

- [ ] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [x] documentation is changed or added
- [ ] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -195,10 +195,11 @@ function getToken (req) {
 }
 ```
 
-We must use the `preHandler` lifecycle if we want to accept CSRF tokens
-inside request bodies. The default property matched is `_csrf`, but it's possible
-to constumize this by chainging the `getToken` function.
-If we are using headers, using the `onRequest` lifecycle is sufficient and safer.
+It is recommended to provide a custom `getToken` function for security and performance reasons.
```
Comment (Eomm) [inline README.md:198]: I would link some OWASP documentation like 

https://cheatsheetseries.owasp.org/cheatsheets/Cross-Site_Request_Forgery_Prevention_Cheat_Sheet.html#use-of-custom-request-headers

Comment (Uzlopak) [reply README.md:198]: Like this?

```suggestion
It is recommended to provide a custom `getToken` function for performancs and [security](https://cheatsheetseries.owasp.org/cheatsheets/Cross-Site_Request_Forgery_Prevention_Cheat_Sheet.html#use-of-custom-request-headers) reasons.
```

Comment (Uzlopak) [reply README.md:198]: @Eomm

Comment (Uzlopak) [reply README.md]: Done

**Review by Eomm — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -15,7 +15,7 @@ We do not claim that this module is able to protect an application without a cle
 @fastify/csrf-protection provides a series of utilities that developers can use to secure their application.
 We recommend using [@fastify/helmet](https://github.com/fastify/fastify-helmet) to implement some of those mitigations.
 
-Security is always a tradeoff between risk mitigation, functionality, and developer experience.
+Security is always a tradeoff between risk mitigation, functionality, performance and developer experience.
```
Comment (Fdawgs) [inline README.md:18]:
```suggestion
Security is always a tradeoff between risk mitigation, functionality, performance, and developer experience.
```

File: README.md [hunk 3]
```diff
@@ -165,11 +165,11 @@ This option is needed to protect against cookie tossing.
 ### `fastify.csrfProtection(request, reply, next)`
 
 A hook that you can use for protecting routes or entire plugins from CSRF attacks.
-Generally, we recommend to use the `onRequest` hook, but if you are sending the token
-via the body, then you should use `preValidation` or `preHandler`.
+Generally, we recommend to use a `onRequest` hook, but if you are sending the token
```
Comment (Fdawgs) [inline README.md:168]:
```suggestion
Generally, we recommend using an `onRequest` hook, but if you are sending the token
```

**Review by Fdawgs — CHANGES_REQUESTED** [review]

**Review by Fdawgs — APPROVED** [review]

**Comment by Uzlopak** [comment]
Thank you @Fdawgs

**Pull request merged**